src/Nethermind/Nethermind.Blockchain/BeaconBlockRoot/BeaconBlockRootHandler.cs

```diff
@@ -1,38 +1,57 @@
 // SPDX-FileCopyrightText: 2023 Demerzel Solutions Limited
 // SPDX-License-Identifier: LGPL-3.0-only
 
-using Nethermind.Core.Specs;
+using System;
 using Nethermind.Core;
-using Nethermind.Int256;
-using Nethermind.State;
 using Nethermind.Core.Crypto;
 using Nethermind.Core.Extensions;
+using Nethermind.Core.Specs;
+using Nethermind.Crypto;
+using Nethermind.Evm.Tracing;
+using Nethermind.Evm.TransactionProcessing;
+using Nethermind.Int256;
+using Nethermind.Logging;
+using Nethermind.State;
 
-namespace Nethermind.Consensus.BeaconBlockRoot;
+namespace Nethermind.Blockchain.BeaconBlockRoot;
 public class BeaconBlockRootHandler : IBeaconBlockRootHandler
 {
-    public static UInt256 HISTORICAL_ROOTS_LENGTH = 8191;
-    private static readonly Address DefaultPbbrContractAddress = new Address("0x000F3df6D732807Ef1319fB7B8bB8522d0Beac02");
-
-    public void ApplyContractStateChanges(Block block, IReleaseSpec spec, IWorldState stateProvider)
+    private readonly ITransactionProcessor _processor;
+    private static Address Default4788Address = new Address("0x000f3df6d732807ef1319fb7b8bb8522d0beac02");
+    private readonly ILogger _logger;
+    private const long GasLimit = 30_000_000L;
+    public BeaconBlockRootHandler(
+        ITransactionProcessor processor,
+        ILogManager logManager)
+    {
+        _processor = processor;
+        _logger = logManager.GetClassLogger();
+    }
+    public void ExecuteSystemCall(Block block, IReleaseSpec spec)
     {
+        BlockHeader? header = block.Header;
         if (!spec.IsBeaconBlockRootAvailable ||
-            block.IsGenesis ||
-            block.Header.ParentBeaconBlockRoot is null) return;
-        Address? eip4788Account = spec.Eip4788ContractAddress ?? DefaultPbbrContractAddress;
-        if (!stateProvider.AccountExists(eip4788Account))
-            return;
-
-        UInt256 timestamp = (UInt256)block.Timestamp;
-        Hash256 parentBeaconBlockRoot = block.ParentBeaconBlockRoot;
-
-        UInt256.Mod(timestamp, HISTORICAL_ROOTS_LENGTH, out UInt256 timestampReduced);
-        UInt256 rootIndex = timestampReduced + HISTORICAL_ROOTS_LENGTH;
+            header.IsGenesis ||
+            header.ParentBeaconBlockRoot is null) return;
 
-        StorageCell tsStorageCell = new(eip4788Account, timestampReduced);
-        StorageCell brStorageCell = new(eip4788Account, rootIndex);
+        Transaction? transaction = new()
+        {
+            Value = UInt256.Zero,
+            Data = header.ParentBeaconBlockRoot.Bytes.ToArray(),
```

Comment:
This just works? Don't we need to do a proper call with ABI coded parameters?

Reply:
yeah, no need for ABI since it's a simple smart contract